### PR TITLE
Added Enum, Optimized / Enhance functions

### PR DESCRIPTION
**Pull request by Arvmor**
- Removed Unused imports
- Player Status Enum (Alive / dead / paused ...)
- More idiomatic structure
-

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,80 +1,104 @@
...
+use std::{cmp::Ordering::*, io::{stdout, Stdout, Write}, time::Duration};
 use std::{thread, time};
 use rand::{thread_rng, Rng};
 
 use crossterm::{
-    cursor::{Hide, MoveTo, Show}, event::{self, poll, read, Event, KeyCode}, execute, style::{Color, Print, ResetColor, SetBackgroundColor, SetForegroundColor}, terminal::{disable_raw_mode, enable_raw_mode, size, Clear}, ExecutableCommand, QueueableCommand
+    cursor::{Hide, MoveTo, Show}, event::{poll, read, Event, KeyCode}, style::Print, terminal::{disable_raw_mode, enable_raw_mode, size, Clear}, ExecutableCommand, QueueableCommand
 };
 
+#[derive(PartialEq, Eq)]
```
Comment (Arvmor) [inline src/main.rs:9]: You can use Default drive for Alive, 
might confuse beginners!

File: src/main.rs [hunk 2]
```diff
@@ -1,80 +1,104 @@
...
+    sc.queue(MoveTo(world.player_c, world.player_l))?
+        .queue(Print("P"))?
+        .flush()?;
 
     Ok(())
 }
 
 
-fn physics(mut world: World) -> Result<World>{
+fn physics(world: &mut World) {
```
Comment (Arvmor) [inline src/main.rs:65]: it doesn't need to return the struct, a mutable reference does the job.

File: src/main.rs [hunk 3]
```diff
@@ -1,80 +1,104 @@
...
-    if world.next_right > world.map[0].1 {
-        world.map[0].1 += 1;
+    for l in (1..world.map.len()).rev() {
+        world.map[l] = world.map[l - 1];
     }
-    if world.next_right < world.map[0].1 {
-        world.map[0].1 -= 1;
-    } 
+
+    let (left, right) = &mut world.map[0];
```
Comment (Arvmor) [inline src/main.rs:79]: Using match to handle the situations

File: src/main.rs [hunk 4]
```diff
@@ -1,80 +1,104 @@
...
     // TODO: below randoms may 1) go outside of range
     if world.next_left == world.map[0].0 && rng.gen_range(0..10) >= 7 {
         world.next_left = rng.gen_range(world.next_left-5..world.next_left+5)
     }
     if world.next_right == world.map[0].1 && rng.gen_range(0..10) >= 7  {
         world.next_right = rng.gen_range(world.next_right-5..world.next_right+5)
     }
-    if world.next_right - world.next_left < 3 { // todo: check abs
+
+    if world.next_right.abs_diff(world.next_left) < 3 {
```
Comment (Arvmor) [inline src/main.rs:99]: using absolute

File: src/main.rs [hunk 5]
```diff
@@ -1,80 +1,104 @@
...
+    cursor::{Hide, MoveTo, Show}, event::{poll, read, Event, KeyCode}, style::Print, terminal::{disable_raw_mode, enable_raw_mode, size, Clear}, ExecutableCommand, QueueableCommand
 };
 
+#[derive(PartialEq, Eq)]
+enum PlayerStatus {
+    Dead,
+    Alive,
+    Animation,
+    Paused
+}
```
Comment (Arvmor) [inline src/main.rs:15]: You can use Default drive for Alive,
might confuse beginners!

File: src/main.rs [hunk 6]
```diff
@@ -1,80 +1,104 @@
...
+            map: vec![(maxc/2-5, maxc/2+5); maxl as usize],
+            maxc,
+            maxl,
+            status: PlayerStatus::Alive,
+            next_left: maxc / 2 - 7,
+            next_right: maxc / 2 + 7,
+        }
+    }
+
+}
```
Comment (Arvmor) [inline src/main.rs:43]: using implemented functions for structs is advised and more idiomatic

**Comment by jadijadi** [comment]
thanks you very much. great contribution.

**Pull request merged**